src/main/java/com/exactpro/th2/sailfish/utils/filter/CompareFilter.java

```diff
@@ -72,7 +72,7 @@ public CompareFilter(FilterOperation operation, String value) {
     }
 
     @Override
-    public ExpressionResult validate(Object value) throws RuntimeException {
+    protected ExpressionResult validateInternal(Object value) {
         Objects.requireNonNull(value);
```

Comment:
Looks like overhead. It has already been checked for null

Reply:
The super class behavior might change. But we need to make sure the the value parameter is not `null`

Reply:
We can mark method `validate` as final and it will be used as a contract

Reply:
It does not protect us from changing the method in superclass) The extra check is always better than no check at all